addons/account/models/partner.py

```diff
@@ -228,7 +228,10 @@ def _get_fiscal_position(self, partner, delivery=None):
             delivery = partner
 
         # partner manually set fiscal position always win
-        manual_fiscal_position = delivery.property_account_position_id or partner.property_account_position_id
+        manual_fiscal_position = (
+            delivery.with_company(self.env.company).property_account_position_id
+            or partner.with_company(self.env.company).property_account_position_id
```

Comment:
Shorter, and if for some reason the definition of `company` changes, we don't need to change it here too
```suggestion
            delivery.with_company(company).property_account_position_id
            or partner.with_company(company).property_account_position_id
```